tests/test_equivalence.py

```diff
@@ -20,13 +21,24 @@
 from fasm2bels.fasm2bels import main
 
 
-test_names = ["add32"]
+test_names = ["add32","alu"]
 
 
 def unpack_tar(tar_file):
     tar = tarfile.open(name=tar_file, mode="r:gz")
     tar.extractall(path=os.path.dirname(tar_file))
 
+def create_golden_file(file_name):
+    vivado = os.environ.get('VIVADO_PATH', None) 
+    assert vivado is not None, "VIVADO_PATH enviromental variable was not set! Please use export VIVADO_PATH=(path to vivado)"
+    file_exists = exists("equivalence_checking_data/"+file_name+"/top_bit.golden.v") 
+    if(file_exists):
+        subprocess.run(["echo", "Golden File already exists."])
+    else:
+        temp = tempfile.TemporaryDirectory()
+        subprocess.run([vivado, "-mode", "batch", "-source", "create_golden_file.tcl", "-tclargs", file_name, temp.name])
+        subprocess.run(["rm", "vivado.log", "vivado.jou", "-r", ".Xil"])    
+        temp.cleanup()
```

Comment:
I think that it would be better to just have the following:

- temporary directory created
- all files generated by vivado need to be put on the temp directory
- the temporary directory is passed to the `create_golden_file` caller
- once the golden file is checked later on in the test, the temporary directory is cleaned up with `temp.cleanup()`